modules/currency-l1/src/main/scala/org/tessellation/currency/l1/http/Routes.scala

```diff
@@ -1,29 +1,61 @@
 package org.tessellation.currency.l1.http
 
+import cats.data.Validated.{Invalid, Valid}
 import cats.effect.Async
 import cats.effect.std.{Queue, Supervisor}
 import cats.syntax.all._
 
-import org.tessellation.currency.DataUpdate
+import org.tessellation.currency.l1.domain.error.{InvalidDataUpdate, InvalidSignature}
+import org.tessellation.currency.{BaseDataApplicationL1Service, DataUpdate}
 import org.tessellation.dag.l1.domain.dataApplication.consensus.ConsensusInput
+import org.tessellation.ext.http4s.error._
+import org.tessellation.kryo.KryoSerializer
 import org.tessellation.sdk.domain.cluster.storage.L0ClusterStorage
+import org.tessellation.security.SecurityProvider
 import org.tessellation.security.signature.Signed
 
 import io.circe.Decoder
+import io.circe.shapes._
 import org.http4s.HttpRoutes
-import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+import org.http4s.circe.CirceEntityCodec.{circeEntityDecoder, circeEntityEncoder}
 import org.http4s.dsl.Http4sDsl
 import org.typelevel.log4cats.slf4j.Slf4jLogger
+import shapeless._
+import shapeless.syntax.singleton._
 
-final case class Routes[F[_]: Async](
+final case class Routes[F[_]: Async: KryoSerializer: SecurityProvider](
   dataApplicationPeerConsensusInput: Queue[F, Signed[ConsensusInput.PeerConsensusInput]],
   l0ClusterStorage: L0ClusterStorage[F],
-  dataDecoder: Decoder[DataUpdate]
+  dataDecoder: Decoder[DataUpdate],
```

Comment:
You can make it implicit and get rid of line `39`. We rarely pass decoders explicitly.

Reply:
I don't want to pass it implicitly as anyway I need to pass it explicitly to the next decoder. I know we pass all decoders/encoders implicitly, but here the decoder is provided by the client (end-user), hence it's passed explicitly as an argument.

Reply:
anyway, I see that I pass also the data application service, so I can extract it from there.